src/components/UpcomingQuizCard/icon/CalendarIcon.tsx

```diff
@@ -0,0 +1,58 @@
+import { Box, type SxProps, type Theme } from '@mui/material';
+import React from 'react';
+
+export interface CalendarIconProps {
+  day?: number | string;
+  fontSize?: 'small' | 'medium' | 'large';
+  color?: string;
+  sx?: SxProps<Theme>;
+}
+
+const CalendarIcon: React.FC<CalendarIconProps> = ({
+  day = new Date().getDate(),
+  fontSize = 'medium',
+  color = '#1976d2',
+  sx,
+}) => {
+  const sizeMap = {
+    small: 32,
+    medium: 48,
+    large: 64,
+  };
+
+  const size = sizeMap[fontSize] || 48;
+
+  return (
+    <Box component="span" sx={{ display: 'inline-flex', alignItems: 'center', ...sx }}>
+      <svg width={size} height={size} viewBox="0 0 64 64" xmlns="http://www.w3.org/2000/svg">
+        <rect
+          x="4"
+          y="8"
+          width="56"
+          height="52"
+          rx="8"
+          ry="8"
+          fill="#f0f0f0"
```

Comment:
Several SVG fill/stroke values are hard-coded (e.g., `fill="#f0f0f0"`, `stroke="#ccc"`). These should be derived from the theme (palette) or use `currentColor` so that color can be controlled by parent components and theme.

**Sugerencia:**
```typescript
Use theme tokens or `currentColor` for SVG shapes. Example: `fill="currentColor"` and set color via the Box sx: `<Box sx={{ color: theme.palette.background.default }}>` or pass explicit theme-derived fills from parent.
```